test/int4_test.cpp

```diff
@@ -0,0 +1,115 @@
+        auto conv = m1.add_instruction(
+            migraphx::make_op("convolution",
+                              {{"padding", {1, 1}}, {"stride", {2, 2}}, {"dilation", {1, 1}}}),
+            x,
+            w);
+        m1.add_instruction(migraphx::make_op("relu"), conv);
+    }
+    migraphx::run_passes(m1, {migraphx::quantize_int4_pass{}});
+
+    auto chk_1 = match::name("quantizelinear")(
+                     match::output(match::name("pack_int4")(match::output(match::name(
+                         "unpack_int4")(match::output(match::name("dequantizelinear")))))))
+                     .bind("q");
+
+    auto res = find_match(m1, chk_1);
+
+    EXPECT(migraphx::contains(res.instructions, "q"));
+}
+
+TEST_CASE(int4_const_prop_test)
+{
+    migraphx::module m1;
+    {
+        auto x = m1.add_parameter("x", {migraphx::shape::float_type, {1, 8, 16, 16}});
+        auto w = m1.add_literal(
+            migraphx::generate_literal({migraphx::shape::float_type, {16, 8, 6, 6}}));
+        auto conv = m1.add_instruction(
+            migraphx::make_op("convolution",
+                              {{"padding", {1, 1}}, {"stride", {2, 2}}, {"dilation", {1, 1}}}),
+            x,
+            w);
+        m1.add_instruction(migraphx::make_op("relu"), conv);
+    }
+    migraphx::run_passes(m1,
+                         {migraphx::quantize_int4_pass{},
+                          migraphx::propagate_constant{},
+                          migraphx::dead_code_elimination{}});
+
+    auto chk_1 = match::name("pack_int4").bind("pack_int4");
+    auto res_1 = find_match(m1, chk_1);
+    EXPECT(not migraphx::contains(res_1.instructions, "pack_int4"));
+
+    auto chk_2 = match::name("unpack_int4").bind("unpack_int4");
+    auto res_2 = find_match(m1, chk_2);
+    EXPECT(migraphx::contains(res_2.instructions, "unpack_int4"));
+}
+
+TEST_CASE(int4_simplify_qdq_pass_test)
+{
+    migraphx::module m1;
+    {
+        auto x        = m1.add_parameter("x", {migraphx::shape::float_type, {1, 8, 16, 16}});
+        auto scale    = m1.add_parameter("sc", {migraphx::shape::float_type, {1, 8, 16, 16}});
+        auto zp       = m1.add_parameter("zp", {migraphx::shape::int8_type, {1, 8, 16, 8}});
+        auto un_pk_zp = m1.add_instruction(migraphx::make_op("unpack_int4"), zp);
+        auto q        = m1.add_instruction(migraphx::make_op("quantizelinear"), x, scale, un_pk_zp);
+        auto dq = m1.add_instruction(migraphx::make_op("dequantizelinear"), q, scale, un_pk_zp);
+        m1.add_return({dq});
+    }
+
+    migraphx::run_passes(m1, {migraphx::simplify_qdq{}});
+
+    auto chk_1 = match::name("quantizelinear")(
+                     match::output(match::name("pack_int4")(match::output(match::name(
+                         "unpack_int4")(match::output(match::name("dequantizelinear")))))))
+                     .bind("q");
+    auto res_1 = find_match(m1, chk_1);
+    EXPECT(migraphx::contains(res_1.instructions, "q"));
+}
```

Comment:
This should be moved with the other simplify_qdq tests, but lets do that later. I think this should just be merged in now.